packages/jest-dom-mocks/src/location.ts

```diff
@@ -24,9 +24,9 @@ export default class Location {
 
     this.originalAssign = window.location.assign;
     this.originalReload = window.location.reload;
-    window.location.assign = jest.fn();
-    window.location.reload = jest.fn();
-    window.location.replace = jest.fn();
+    jest.spyOn(window.location, 'assign');
+    jest.spyOn(window.location, 'reload');
+    jest.spyOn(window.location, 'replace');
```

Comment:
If we are doing it this way we don't need the assignment of the original versions, we can just restore after we are done